### PR TITLE
8276847: JSException: ReferenceError: Can't find variable: IntersectionObserver

### PR DESCRIPTION
**Pull request by jperedadnr**
This PR enables intersection observer support for WebKit, and includes a system test.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276847](https://bugs.openjdk.java.net/browse/JDK-8276847): JSException: ReferenceError: Can't find variable: IntersectionObserver

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.java.net/census#arapte) (@arapte - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/664/head:pull/664` \
`$ git checkout pull/664`

Update a local copy of the PR: \
`$ git checkout pull/664` \
`$ git pull https://git.openjdk.java.net/jfx pull/664/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 664`

View PR using the GUI difftool: \
`$ git pr show -t 664`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/664.diff">https://git.openjdk.java.net/jfx/pull/664.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
This looks like a reasonable thing to do.

/reviewers 2

**Comment by jperedadnr** [comment]
Worth pointing out that [JDK-8273558](https://bugs.openjdk.java.net/browse/JDK-8273558) gets fixed with this PR.

File: tests/system/src/test/resources/test/javafx/scene/web/testIObserver.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        }
+    </script>
+</head>
+<body>
+<div id="parent">
+    <div id="target"></div>
+</div>
+<div id="output">Intersection ratio: <pre>?</pre></div>
+</body>
+</html>
```
Comment (kevinrushforth) [inline tests/system/src/test/resources/test/javafx/scene/web/testIObserver.html:54]: Minor: missing newline at end of file

**Review by kevinrushforth — APPROVED** [review]
Looks good. I've tested this on all platforms and confirm that it fixes the issue with no regressions that I can see. I also confirm that the new test catches the problem (fails without and passes with the fix), and that this fixes [JDK-8273558](https://bugs.openjdk.java.net/browse/JDK-8273558), which can be closed as a duplicate.

I left one minor formatting comment, but unless there are other changes, it probably isn't worth changing it.

**Review by arapte — APPROVED** [review]
Looks good to me. Shall re-approve when PR is updated as per Kevin's suggestion.

**Comment by kevinrushforth** [comment]
> Looks good to me. Shall re-approve when PR is updated as per Kevin's suggestion.

@jperedadnr My suggestion was optional, so you can either integrate it as-is or make the (minor) change and we'll re-approve.

**Comment by jperedadnr** [comment]
No problem, I just pushed the minor change.

**Review by kevinrushforth — APPROVED** [review]

**Review by arapte — APPROVED** [review]

**Pull request closed without merging**